mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -1821,6 +1824,22 @@ impl TendermintCoin {
             _ => (self.gas_price(), fallback_gas_limit),
         }
     }
+
+    fn active_ticker_and_decimals_from_denom(&self, denom: &str) -> Option<(String, u8)> {
+        if self.denom.to_string() == denom {
+            return Some((self.ticker.clone(), self.decimals));
+        }
+
+        let tokens = self.tokens_info.lock();
+
+        for (ticker, token) in &*tokens {
+            if token.denom.to_string() == denom {
```

Comment:
`denom.as_ref()` can be used instead of `to_string()`. Also, is it possible to reorganize data structure in some way to avoid iterating over all tokens here?

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/1978/commits/5464064251e0fc1334e628dc42562a4a4cce8977